### PR TITLE
Update nidigital API methods to use pins repeated capability

### LINKED ISSUE #1294: Update methods in nidigital API to be Pins repeated-capability-based, where applicable
### Description of issue

There methods should be updated such that `pins` repeated capability can be used on them:
- configure_time_set_compare_edges_strobe
- configure_time_set_compare_edges_strobe2x
- configure_time_set_drive_edges
- configure_time_set_drive_edges2x
- configure_time_set_drive_format
- configure_time_set_edge
- configure_time_set_edge_multiplier
- create_capture_waveform_parallel
- create_capture_waveform_serial
- create_source_waveform_parallel
- create_source_waveform_serial
- fetch_history_ram_cycle_information
- get_time_set_drive_format
- get_time_set_edge
- get_time_set_edge_multiplier

### PR DESCRIPTION
**Pull request by sbethur**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

- [x] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.

- [x] I've added tests applicable for this pull request

### What does this Pull Request accomplish?

Following methods are updated such that pins repeated capability can be used on them:
```
configure_time_set_compare_edges_strobe
configure_time_set_compare_edges_strobe2x
configure_time_set_drive_edges
configure_time_set_drive_edges2x
configure_time_set_drive_format
configure_time_set_edge
configure_time_set_edge_multiplier
create_capture_waveform_parallel
create_capture_waveform_serial
create_source_waveform_parallel
create_source_waveform_serial
fetch_history_ram_cycle_information
get_time_set_drive_format
get_time_set_edge
get_time_set_edge_multiplier
```

### List issues fixed by this Pull Request below, if any.

* Fix #1294 

### What testing has been done?

A new system test is added to verify pins can be used as repeated capability.

### CODE REVIEW

File: generated/nidigital/nidigital/session.py [hunk 1]
```diff
@@ -1011,226 +1011,401 @@ def _get_attribute_vi_string(self, attribute):
...
+        bit_order_ctype = _visatype.ViInt32(bit_order)  # case S150
+        error_code = self._library.niDigital_CreateSourceWaveformSerial(vi_ctype, pin_list_ctype, waveform_name_ctype, data_mapping_ctype, sample_width_ctype, bit_order_ctype)
         errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
-        return name_ctype.value.decode(self._encoding)
+        return
 
     @ivi_synchronized
-    def _get_pin_results_pin_information(self):
-        r'''_get_pin_results_pin_information
+    def fetch_history_ram_cycle_information(self, site, position, samples_to_read):
```
Comment (sbethur) [inline generated/nidigital/nidigital/session.py]: Just realized that in #1111, `sites` is being made into a repeated capability. We need to choose one over the other.

File: docs/nidigital/class.rst [hunk 2]
```diff
@@ -428,21 +428,23 @@ configure_time_set_compare_edges_strobe
...
+                nidigital.Session object, then the method will use all repeated capabilities in the session.
+                You can specify a subset of repeated capabilities using the Python index notation on an
+                nidigital.Session repeated capabilities container, and calling this method on the result.:
 
-            :param pin_list:
-
+                .. code:: python
 
-                
+                    session.pins[0,1].configure_time_set_compare_edges_strobe(time_set, strobe_edge)
```
Comment (marcoskirsch) [inline docs/nidigital/class.rst]: Shreyas, what do you think of using something better for this example, like "VCC"?
Should be a separate PR/issue though, and it's low priority.

Comment (sbethur) [reply docs/nidigital/class.rst]: Issue already present - #1125

File: generated/nidigital/nidigital/session.py (hunk 1 above)
Comment (sbethur) [reply generated/nidigital/nidigital/session.py]: Will be handled in [this](https://github.com/ni/nimi-python/pull/1290/files#r387789387) PR.
